client/src/main/java/org/spine3/type/TypeUrl.java

```diff
@@ -230,8 +257,25 @@ private static String getTypeUrlPrefix(GenericDescriptor descriptor) {
      * @throws UnknownTypeException wrapping {@link ClassNotFoundException} if
      *         there is no corresponding Java class
      */
-    public <T extends Message> Class<T> toMessageClass() {
-        return KnownTypes.getClass(this);
+    public <T extends Message> Class<T> getJavaClass() throws UnknownTypeException {
+        return KnownTypes.getJavaClass(this);
+    }
+
+    /**
+     * Obtains the type descriptor for the type of this URL.
+     *
+     * @return {@code Descriptor} or {@code Optional.absent()} if this type URL
+     *         represents an outer class
+     */
+    public Optional<Descriptor> getTypeDescriptor() {
+        final Class<? extends Message> clazz = getJavaClass();
+        final GenericDescriptor descriptor = getClassDescriptor(clazz);
+        // Skip outer class descriptors.
+        if (descriptor instanceof Descriptor) {
```

Comment:
- Outer class descriptors are represented with `FileDescriptor` type.
 - There are concrete type descriptors that are not represented with `Descriptor` type (e.g. `enum`s' descriptors are `EnumDescriptor extends GenericDescriptor`).

If we want to avoid __outer class__ descriptors, we should check `!(descriptor instanceof FileDescriptor)` instead.